### ISSUE TITLE
Question about DyLinear block

### ISSUE THREAD

**Issue by khchul** [mims-harvard/UniTS#7]
Thanks for the great work!
I've read your paper and got confused in DyLinear blocks and ask here to get some clarifications.

Q, Is DyLinear block applied to each variables independently? In equation (2), I believe that $z_s$ is sized $l_s$ * v * d and W<sub>Interp</sub> is sized $w_i * w_o$. How are W<sub>Interp</sub>$z_s$ calculated and what is exactly l<sub>out</sub> ? Also if it is calculated as independent, why do we need interpolation? Isn't d fixed among different time series?

Thank you

**Comment by gasvn** [comment]
Yes,  DyLinear block applied to each variables independently. Dylinear is used to change to weight shape to make it match various sequence lengths.

**Comment by khchul** [comment]
Thank you for your clarification!
Though I still can't follow the purpose of DyLinear block. 
I first thought it was added to force the variable sized sequences to have fixed length, but realized this cannot be done due to the division of Z<sub>in</sub> (since Z<sup>2</sup><sub>mid</sub> doesn't pass DyLinear operator).

Could you elaborate more of the effect of DyLinear block?
Thank you
